### PR TITLE
RE2: parser fixing; workaround

### PR DESCRIPTION
**Pull request by vhavlena**
Currently it seems RE2 parser has a problem with removing epsilons (the epsilon removal is somehow hardcoded in the parser). This PR disables the internal epsilon removal and forces to use the mata function instead. Also adds a new parameter to the parser allowing to reduce the resulting automaton (it might be useful). Related to #139

### CODE REVIEW

File: src/re2parser.cc [hunk 1]
```diff
@@ -562,16 +562,23 @@ namespace {
  * @param epsilon_value value, that will represent epsilon on transitions
  * @return mata::Nfa::Nfa corresponding to pattern
  */
-void Mata::RE2Parser::create_nfa(Nfa::Nfa* nfa, const std::string& pattern, bool use_epsilon, int epsilon_value) {
+void Mata::RE2Parser::create_nfa(Nfa::Nfa* nfa, const std::string& pattern, bool use_epsilon, int epsilon_value, bool use_reduce) {
```
Comment (Adda0) [inline src/re2parser.cc]: Parameter `use_reduce` could benefit from a documentation string explaining that is uses reduce on the result as well as trimming the result beforehand. This is just an idea, totally optional, though.

**Review by Adda0 — APPROVED** [review]
LGTM. We have already discussed this workaround for #139 and as of now, it is the simplest solution to give us a working RE2 parser.

File: src/re2parser.cc (hunk 1 above)
Comment (vhavlena) [reply src/re2parser.cc]: ok

File: src/re2parser.cc [hunk 2]
```diff
@@ -560,18 +560,26 @@ namespace {
...
     if (nfa == NULL) {
         throw std::runtime_error("create_nfa: nfa should not be NULL");
     }
 
     RegexParser regexParser{};
     auto parsed_regex = regexParser.parse_regex_string(pattern);
     auto program = parsed_regex->CompileToProg(regexParser.options.max_mem() * 2 / 3);
-    regexParser.convert_pro_to_nfa(nfa, program, use_epsilon, epsilon_value);
+    regexParser.convert_pro_to_nfa(nfa, program, true, epsilon_value);
```
Comment (tfiedor) [inline src/re2parser.cc:574]: Is this change intended? Seems quite out of place wrt rest of the changes.

**Review by tfiedor — APPROVED** [review]
Seems OK.

File: src/re2parser.cc (hunk 2 above)
Comment (Adda0) [reply src/re2parser.cc:574]: It is. The workaround is to always use epsilon (hence the change from `use_epsilon` to `true`) and remove epsilon symbols afterwards, if required.

Comment (tfiedor) [reply src/re2parser.cc:574]: I would then add comment explaining this. Since someone might in future come here, see the true and change it o `use_epsilon` thinking it is mistake.

Comment (vhavlena) [reply src/re2parser.cc:574]: Yes, I changed it deliberately. It seems that inside `convert_pro_to_nfa` the code removing epsilons is not working properly (if `use_epsilon` is set to true, the problematic code is not used).

Comment (vhavlena) [reply src/re2parser.cc:574]: I suppose we remove the problematic code from `convert_pro_to_nfa` entirely at some point. So this parameter will not be used anymore

**Review by martinhruska — APPROVED** [review]
I have no issues with PR. If everyone approves it, I'm gonna merge it.

**Review by jurajsic — APPROVED** [review]
